components/org.wso2.carbon.identity.governance/src/main/java/org/wso2/carbon/identity/governance/internal/service/impl/notification/DefaultNotificationChannelManager.java

```diff
@@ -0,0 +1,351 @@
+        // channel list. Therefore, sending the first channel as in the list as preferred channel.
+        String availableChannel = availableChannels.get(0);
+        if (log.isDebugEnabled()) {
+            String message = String.format("User does not have server default channel : %1$s. Therefore,"
+                    + "communication channel is set to : %2$s", serverDefaultChannel, availableChannel);
+            log.debug(message);
+        }
+        return availableChannel;
+    }
+
+    /**
+     * Create a claims list related to notification channels using the notification channel objects.
+     *
+     * @param channels Notification channel objects
+     * @return List of claims associated with the notification channels
+     */
+    private ArrayList<String> getChannelsClaimList(ArrayList<NotificationChannels> channels) {
+
+        ArrayList<String> claimList = new ArrayList<>();
+        for (NotificationChannels channel : channels) {
+            claimList.add(channel.getClaimUri());
+        }
+        return claimList;
+    }
+
+    /**
+     * Build the user claims map with notification channel information.
+     *
+     * @param username     Username
+     * @param tenantDomain Tenant domain
+     * @return User claims map with notification channel information
+     * @throws NotificationChannelManagerServerException Error while getting user claims
+     */
+    private Map<String, String> buildChannelClaimsMap(String username, String tenantDomain)
+            throws NotificationChannelManagerServerException {
+
+        ArrayList<String> claimUrls = getChannelsClaimList(this.channels);
+        claimUrls.add(IdentityMgtConstants.Claim.PREFERED_CHANNEL_CLAIM);
+        UserStoreManager userStoreManager = getUserstoreManager(username, tenantDomain);
+        Map<String, String> claimValues;
+        try {
+            claimValues = userStoreManager.getUserClaimValues(username, claimUrls.toArray(new String[0]), null);
+        } catch (UserStoreException exception) {
+
+            // This error will occur due to unavailability of identity claims.
+            // Retrieving email address and mobile claims for user.
+            String[] channelClaims = new String[] {
+                    NotificationChannels.SMS_CHANNEL.getClaimUri(), NotificationChannels.EMAIL_CHANNEL.getClaimUri()
+            };
+            try {
+                claimValues = userStoreManager.getUserClaimValues(username, channelClaims, null);
+            } catch (UserStoreException e) {
+                throw new NotificationChannelManagerServerException(
+                        IdentityMgtConstants.ErrorMessages.ERROR_CODE_DEFAULT_SERVER_ERROR.getCode(),
+                        IdentityMgtConstants.ErrorMessages.ERROR_CODE_DEFAULT_SERVER_ERROR.getMessage(), e);
+            }
+        }
+        return claimValues;
+    }
+
+    /**
+     * Get the UserStoreManager
```

Comment:
As a best practice, the description should end with a full stop. Check this throughout the PR

Reply:
Fixed by https://github.com/wso2-extensions/identity-governance/pull/342